### PR TITLE
clarify registered vs facts

### LINKED ISSUE #45411: Registered variable is not delegated as a fact
##### SUMMARY

In the docs, it mentions:

> Registered variables are valid on the host the remainder of the playbook run, which is the same as the lifetime of “facts” in Ansible. Effectively registered variables are just like facts.

But if `delegate_facts` is set to `yes` in task that registers a variable, then the variable is not available in the `hostvars` for the host that actually performed the task through delegation.

##### ISSUE TYPE
- ~Bug  Report~
- Docs Report

##### COMPONENT NAME

delegate_facts

##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0 (devel 6110163f99) last updated 2018/09/10 08:39:48 (GMT +200)
  config file = None
  configured module search path = ['/home/nikos/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nikos/Projects/ansible/lib/ansible
  executable location = /home/nikos/Projects/ansible/bin/ansible
  python version = 3.7.0 (default, Jul 15 2018, 10:44:58) [GCC 8.1.1 20180531]
```

##### CONFIGURATION

```paste below
Nothing
```

##### OS / ENVIRONMENT

Arch linux

##### STEPS TO REPRODUCE

```yaml
- hosts: server                         
  gather_facts: no                      
  tasks:                                
    - set_fact:                         
        qqq: 111                        
    - set_fact:                         
        ggg: 222                        
      delegate_to: localhost            
      delegate_facts: yes               
    - debug:                            
        var: qqq                        
      register: result                  
      delegate_to: localhost            
      delegate_facts: yes               
    - debug:                            
        var: ggg                        
      register: result                  
      delegate_to: localhost            
      delegate_facts: yes               
    - debug:                            
        var: hostvars['localhost']      
```

##### EXPECTED RESULTS

I expected to see the variable `result` in the output of the last debug.

##### ACTUAL RESULTS

```paste below
PLAY [server] ****************************************************************************************************************************************************************************************************

TASK [set_fact] **************************************************************************************************************************************************************************************************
ok: [server]

TASK [set_fact] **************************************************************************************************************************************************************************************************
ok: [server -> localhost]

TASK [debug] *****************************************************************************************************************************************************************************************************
ok: [server -> localhost] => {
    "qqq": 111
}

TASK [debug] *****************************************************************************************************************************************************************************************************
ok: [server -> localhost] => {
    "ggg": "VARIABLE IS NOT DEFINED!"
}

TASK [debug] *****************************************************************************************************************************************************************************************************
ok: [server] => {
    "hostvars['localhost']": {
        "ansible_check_mode": false,
        "ansible_connection": "local",
        "ansible_diff_mode": false,
        "ansible_facts": {},
        "ansible_forks": 5,
        "ansible_inventory_sources": [
            "/home/nikos/Projects/ansible-demo/inventory.yml"
        ],
        "ansible_playbook_python": "/usr/bin/python",
        "ansible_python_interpreter": "/usr/bin/python",
        "ansible_run_tags": [
            "all"
        ],
        "ansible_skip_tags": [],
        "ansible_verbosity": 0,
        "ansible_version": {
            "full": "2.8.0.dev0",
            "major": 2,
            "minor": 8,
            "revision": 0,
            "string": "2.8.0.dev0"
        },
        "ggg": 222,
        "group_names": [],
        "groups": {
            "all": [
                "server"
            ],
            "ungrouped": [
                "server"
            ]
        },
        "inventory_hostname": "localhost",
        "inventory_hostname_short": "localhost",
        "omit": "__omit_place_holder__76e3de3399ce73e0cef35372f695398c8b6d4c55",
        "playbook_dir": "/home/nikos/Projects/ansible-demo"
    }
}
```

Also, although I have noticed it before, it seems a bit unintuitive that `qqq` gets printed (although localhost does not know about it) and `ggg` does not (although localhost knows about it). But this seems like a design decision.

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
fixes #45411

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs
##### ANSIBLE VERSION

```paste below
2.x
```

### CODE REVIEW

**Comment by nkakouros** [comment]
To ask to have registered variables be delegated as facts, is it an overkill?

**Comment by bcoca** [comment]
not something i would do under delegated facts as they are not facts

File: docs/docsite/rst/user_guide/playbooks_variables.rst [hunk 1]
```diff
@@ -604,8 +604,8 @@ While it's mentioned elsewhere in that document too, here's a quick syntax examp
         - shell: /usr/bin/bar
           when: foo_result.rc == 5
 
-Registered variables are valid on the host the remainder of the playbook run, which is the same as the lifetime of "facts"
-in Ansible.  Effectively registered variables are just like facts.
+Registered variables are valid on the host the remainder of the playbook run, which is similar to the lifetime of "facts" in Ansible.
+Both registered variables and facts are host level variables, but they are not the same thing.
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_variables.rst]: How do registered variables differ from facts? Where is a user most likely to run into that difference?

Comment (bcoca) [reply docs/docsite/rst/user_guide/playbooks_variables.rst]: Several ways, but the main ones are:
 - precedence: they follow different priorities 
 - caching: facts are backed by the 'configured cache plugin' vs registered vars are always just kept 'in memory'

**Comment by evrardjp** [comment]
I consider this is already an improvement on the text.

File: docs/docsite/rst/user_guide/playbooks_variables.rst (hunk 1 above)
Comment (robertdebock) [reply docs/docsite/rst/user_guide/playbooks_variables.rst]: Can the suggested text include the precedence and caching details?

The text "which is similar to the lifetime of "facts" in Ansible." does not clarify the validity for me.
```suggestion
Registered variables are valid on the host the remainder of the playbook run.
```

The text "but they are not the same thing." also does not help me.

```suggestion
Both registered variables and facts are host level variables.
```

**Review by samccann — APPROVED** [review]

**Pull request merged**